### PR TITLE
170 - wallet improvements

### PR DESCRIPTION
**Pull request by goranjovic**
added idea 170 - wallet improvements

### CODE REVIEW

**Comment by jeluard** [comment]
@goranjovic I was considering doing this part of the first iterations of `extensions` (which is a prerequisite - or updating our current API). So we can keep this a standalone swarm or figure out how to work together :D

File: ideas/170-wallet-improvements.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    Idea: 170
+    Title: Wallet Improvements
+    Status: Draft
+    Created: 2018-04-13
+    
+## Summary
+
+Status Wallet does not fully support ERC20 tokens (requesting and transaction history not available)
+and generally the transactions UX is different between Chat and Wallet. The intention of this swarm
+is to wrap up those inconsistencies.
```
Comment (chadyj) [inline ideas/170-wallet-improvements.md]: and perhaps add "and study and improve wallet and transaction UX problems and limitations to make the wallet experience great"?

**Comment by goranjovic** [comment]
Sure, there's a lot of overlaps all over the place!

@jeluard Referring to the chat integration specifically or the Wallet changes in general? There is some standalone stuff there already, e.g. token history. Btw, I'd rather not do anything on top of the current API, it's pointless at this stage.

@asemiankevich You mean like, as a user I want to keep track of my token transactions, kitties, etc? Or something more specific that that?

File: ideas/170-wallet-improvements.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+## Swarm Participants
+- Lead Contributor: [@goranjovic](https://github.com/goranjovic)
+- Contributor: 
+- Testing & Evaluation: [@asemiankevich] (https://github.com/asemiankevich)
+- PM: [@chadyj](https://github.com/chadyj)
+- UX:
```
Comment (chadyj) [inline ideas/170-wallet-improvements.md]: @hesterbruikman and @jpbowen can bring huge insight here with UXR!

@denis-sharypin @andmironov are you interested?

File: ideas/170-wallet-improvements.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Description: Integration of Send and Request operations between Wallet and Chat.
+ 
+### Iteration 3
+
+Goal Date: 
+
+Description: ERC721 support in Wallet
+
+Testing days required:
+
```
Comment (chadyj) [inline ideas/170-wallet-improvements.md]: Iteration 4?
UX improvements derived from UXR sessions?

**Comment by jeluard** [comment]
@goranjovic Right chat integration specifically. Missed the other parts after a quick glance.

File: ideas/170-wallet-improvements.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+- Display ERC20 token transfers in transaction history
+- Send ERC20 tokens from chat
+- Request ERC20 tokens from wallet and chat
+- Unified Chat/Wallet transaction UX
+- ERC721 support in Wallet
+
+## Success Metrics
+
+- All exit criteria met
```
Comment (chadyj) [inline ideas/170-wallet-improvements.md]: UXR retested with measurable improvements

File: ideas/170-wallet-improvements.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+## Swarm Participants
+- Lead Contributor: [@goranjovic](https://github.com/goranjovic)
+- Contributor: 
+- Testing & Evaluation: [@asemiankevich] (https://github.com/asemiankevich)
+- PM: [@chadyj](https://github.com/chadyj)
+- UX: 
+
+## Product Overview & Description
+
+Add native support for all ERC20 token operations and rudimentary support for ERC721 tokens.
```
Comment (chadyj) [inline ideas/170-wallet-improvements.md]: Also UXR insight and UX improvements to the wallet experience?

Perhaps a survey similar to messaging as a benchmark that can be used to compare final results?
And specific wallet testing in the upcoming Berlin UXR sessions?

File: ideas/170-wallet-improvements.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+- Possibly depends on extensions swarm for integrations with Chat 
+- Related to #163 - support ERC721 tokens 
+
+
+### Minimum Viable Product
+
+Goal Date: 2018-04-18
+
+Description: Support for ERC20 transaction history in Wallet
```
Comment (chadyj) [inline ideas/170-wallet-improvements.md]: UXR sessions and recommendations would be good to kick off.

**Comment by denis-sharypin** [comment]
hi! we together with @hesterbruikman  wanted to create such swarm.  Can we join this swarm then?

File: ideas/170-wallet-improvements.md (hunk 2 above)
Comment (denis-sharypin) [reply ideas/170-wallet-improvements.md]: @chadyj yep! we together with @hesterbruikman  wanted to create such swarm. We are planning to have a call on Monday. So interested indeed!

**Comment by goranjovic** [comment]
@denis-sharypin @hesterbruikman Welcome to the... team! Or swarm!

I'll pencil you in!

**Comment by chadyj** [comment]
I'd love to see the UXR insight incorporated into this swarm's iteration.

Here are some great ones:

1. Show asset amount
screen shot 2018-04-13 at 3 29 06 pm

2. Transaction escape route
screen shot 2018-04-13 at 3 30 29 pm

3. Insights into transaction status
screen shot 2018-04-13 at 3 31 52 pm
screen shot 2018-04-13 at 3 32 28 pm

**Comment by asemiankevich** [comment]
@goranjovic something like:

1) as a user i want to see my ERC20 transactions in history of transactions (Wallet -> Transaction History)
2) as a user I want to distinguish my ERC20 transactions easily in History of transactions (maybe some design required)
3) as a user i want to understand clearly whom i sent transaction to (ERC20, ETH, etc)
4) as a user i want to request erc20 tokens in 1x1 chat
5) as a user i want to send erc20 tokens in 1x1 chat by responding to appropriate request
6) as a user i want to send erc 20 tokens in a group chat
7) as a user i want to send erc20 tokens in a group chat by responding to appropriate request
maybe @denis-sharypin and @hesterbruikman will add some more user scenarios to scope ?

Also which scenarios do we want to cover for erc721 support?

**Comment by hesterbruikman** [comment]
@asmiankevich , great thinking to list use cases so that we can scope swarm activities. Merging the ones you already wrote and some additions in a sheet. Will add the sheet here asap.

**Comment by hesterbruikman** [comment]
I merged the user stories by @asemiankevich , what came out of earlier UX testing and discussions in this sheet: https://docs.google.com/spreadsheets/d/1z5G4WaFArrvOF6LOUFrY3THzSdFrELPh6vdlGV2BtdU/edit?usp=sharing

**Pull request merged**